aten/src/ATen/native/group_norm.cpp

```diff
@@ -120,10 +120,13 @@ std::tuple<Tensor, Tensor, Tensor> native_group_norm_backward(
       at::borrow_from_optional_tensor(gamma_opt);
   const Tensor& gamma = *gamma_maybe_owned;
 
-  bool mixed_type = is_mixed_type(X, mean, rstd);
+  bool mixed_type = is_mixed_type(X, mean, rstd) || is_mixed_type(dY, mean, rstd);
   if (mixed_type) {
     check_mixed_data_type(X, mean, rstd);
+    check_mixed_data_type(dY, mean, rstd);
   }
+  auto memory_format = X.device().is_cpu() ?
+      X.suggest_memory_format() : at::MemoryFormat::Contiguous;
```

Comment:
Hmm, why just cpu? `xpu` should be the same, shouldn't it?

Reply:
This is because we just add channels last support on CPU.

Reply:
Please correct me if I'm wrong, but as I read the code, this change affects behavior for XPU: it used to just bypass tensor in whatever format it was allocated for CPU and XPU, but with your changes memory layout is converted to a contiguous to contiguous for XPU.

Reply:
`native_group_norm_backward_symint(grads[0].device().is_xpu() ? grads[0] : grads[0].contiguous(), input.device().is_xpu() ? input : input.contiguous(), result1, result2, weight, N, C, HxW, group, grad_input_mask) : std::tuple<Tensor, Tensor, Tensor>())`  

`dX = at::native::empty_like(
        X,
        c10::nullopt /* dtype */,
        c10::nullopt /* layout */,
        c10::nullopt /* device */,
        c10::nullopt /* pin_memory */,
        at::MemoryFormat::Contiguous);`

The original version of GroupNorm backward shown above bypasses grads[0] and input in whatever format it was allocated for XPU, and always generate contiguous dX. 
I want to keep the  behavior for XPU in this PR and also make sure that:  input, grad[0] and dX are memory layout aligned and support both channels last and channels first on CPU.

Reply:
The original version, i.e., before https://github.com/pytorch/pytorch/pull/89485 merged, force the memory format to be contiguous for dX: `dX = at::native::empty_like( X, c10::nullopt /* dtype */, c10::nullopt /* layout */, c10::nullopt /* device */, c10::nullopt /* pin_memory */, at::MemoryFormat::Contiguous);`